packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditGenerators.ts

```diff
@@ -103,26 +114,43 @@ export function viewFromState(
 	initialTree: TreeContent<typeof fuzzSchema.rootFieldSchema>["initialTree"] = undefined,
 ): FuzzView {
 	state.view ??= new Map();
-
-	return (
+	const view =
 		state.transactionViews?.get(client.channel) ??
 		getOrCreate(state.view, client.channel as SharedTree, (tree) => {
+			const treeSchema = treeSchemaFromStoredSchema(
+				(client.channel as SharedTree).storedSchema,
+			);
 			const flexView: FlexTreeView<typeof fuzzSchema.rootFieldSchema> = schematizeFlexTree(
 				tree,
 				{
 					initialTree,
-					schema: fuzzSchema,
+					schema: isEmptyStoredSchema(client.channel as SharedTree)
+						? initialFuzzSchema
+						: treeSchema,
 					allowedSchemaModifications: AllowedUpdateType.Initialize,
 				},
-			);
+				() => {
+					if (state.view?.get(client.channel as SharedTree) !== undefined) {
+						state.view.delete(client.channel as SharedTree);
+					}
+				},
+			) as unknown as FuzzView;
+
 			const fuzzView = flexView as FuzzView;
 			assert.equal(fuzzView.currentSchema, undefined);
-			fuzzView.currentSchema = fuzzNode;
+			const nodeSchema = treeSchema.nodeSchema.get(brand("tree2fuzz.node")) as FuzzNodeSchema;
+			fuzzView.currentSchema =
+				nodeSchema ?? initialFuzzSchema.nodeSchema.get(brand("tree2fuzz.node"));
 			return fuzzView;
-		})
-	);
+		});
+	return view;
 }
 
+function isEmptyStoredSchema(tree: SharedTree): boolean {
+	const rootFieldSchemaData = (tree.storedSchema as unknown as any).rootFieldSchemaData;
+	// TODO: simply using rootFieldSchemaData === storedEmptyFieldSchema does not work. But a better way to make this check should be done.
```

Comment:
Do you know why? is this actionable?

It's not super concerning to me. I don't like having TODOs we don't plan to address hanging around in code though.

Reply:
I checked this again, and still unsure why it does not work. I tested this comparison `rootFieldSchemaData === storedEmptyFieldSchema` in our `onCreate` method (before updating schema) and it seems to work fine, but doesn't seem to work when we call it in our `viewFromState` function. When looking at them from the debugger, I don't see any difference, but still fails the equality.